src/c++/library/http_client.cc

```diff
@@ -48,6 +48,7 @@ extern "C" {
 
 #ifdef _WIN32
 #define strncasecmp(x, y, z) _strnicmp(x, y, z)
+#undef min  // NOMINMAX did not resolve std::min compile error
```

Comment:
from [this stackoverflow question] it seems like explicitly defining the type could also resolve the macro collision issue. Have/Can we try that?

Reply:
We can definitely try that. I think maybe we can fix this at a later stage, because I think we should fix the build stage CI issue first DLIS-4325, and then apply a more concrete fix DLIS-4344.